src/app/app-modules/nurse-doctor/case-sheet/general-case-sheet/doctor-diagnosis-case-sheet/doctor-diagnosis-case-sheet.component.ts

```diff
@@ -320,6 +325,49 @@ export class DoctorDiagnosisCaseSheetComponent
             );
         }
       }
+
+      if (
+        this.visitCategory === 'General OPD (QC)' &&
+        this.casesheetData &&
+        this.casesheetData.doctorData
+      ) {
+        this.referDetails = this.casesheetData.doctorData.Refer;
+        if (this.referDetails?.refrredToAdditionalServiceList) {
+          this.serviceList = this.referDetails.refrredToAdditionalServiceList
+            .map((service: any) => service.serviceName)
+            .filter((name: any) => name !== null && name !== '')
+            .join(',');
+        }
+
+        if (this.referDetails && this.referDetails.referralReason) {
+          console.log('institute', this.referDetails.referralReason);
+          for (let i = 0; i < this.referDetails.referralReason.length; i++) {
+            if (this.referDetails.referralReason[i]) {
+              this.referralReasonList += this.referDetails.referralReason[i];
+              if (i >= 0 && i < this.referDetails.referralReason.length - 1)
+                this.referralReasonList += ',';
+            }
+          }
+        }
+      }
+      console.log(
+        'referDetailsForRefer',
+        JSON.stringify(this.referDetails, null, 4),
+      );
+      if (
+        this.casesheetData &&
+        this.casesheetData.doctorData.Refer &&
+        this.referDetails.revisitDate &&
+        !moment(this.referDetails.revisitDate, 'DD/MM/YYYY', true).isValid()
+      ) {
+        const sDate = new Date(this.referDetails.revisitDate);
```

Comment:
Why would this happen? A date is invalid for moment, but valid for JS Date?

Reply:
moment.js validates dates strictly according to the date format provided (DD/MM/YYYY). Javascript's Date Constructor sometimes accept and parse dates that do not strictly matches with the expected date format.